### PR TITLE
Split qarg, carg in QuantumCircuit and Instruction

### PR DESCRIPTION
**Pull request by ajavadia**
### Summary
As part of cleaning up dag and circuit, and making them compatible, this PR splits the regs in QuantumCircuits into qregs and cregs It also splits the args in Instruction into qargs and cargs. The DAGCircuit class already makes this distinction.

This makes it easier to manipulate the DAG given a QuantumCircuit or some Instructions.

### CODE REVIEW

File: test/python/test_circuit.py [hunk 1]
```diff
@@ -118,8 +118,8 @@ def test_circuit_from_qasm_str(self):
 measure qr2[0] -> cr[1];
 measure qr2[1] -> cr[2];\n"""
         qc = QuantumCircuit.from_qasm_str(qasm_str)
-        self.assertEqual(len(qc.get_cregs()), 1)
```
[the remark this answers is not in the document]
Comment (ajavadia) [reply test/python/test_circuit.py]: Ok I rebased and received your changes.

File: qiskit/extensions/standard/barrier.py [hunk 2]
```diff
@@ -19,35 +19,35 @@ class Barrier(Instruction):
 
     def __init__(self, qubits, circ):
         """Create new barrier instruction."""
-        super().__init__("barrier", [], list(qubits), circ)
+        super().__init__("barrier", [], list(qubits), [], circ)
```
[the remark this answers is not in the document]
Comment (ajavadia) [reply qiskit/extensions/standard/barrier.py]: I don't think so either. In OpenQASM barriers are defined over qargs. I can't think of a scenario right now where barrier over classical may be useful.

File: qiskit/_instruction.py [hunk 3]
```diff
@@ -20,42 +20,48 @@
 
     param: List of parameters to specialize a specific intruction instance.
 
-    arg: List of pairs (Register, index) that the instruction acts on.
+    qargs: List of qubits (QuantumRegister, index) that the instruction acts on.
+
+    cargs: List of clbits (ClassicalRegister, index) that the instruction acts on.
 """
-from sympy import Number, Basic
+import sympy
```
[the remark this answers is not in the document]
Comment (ajavadia) [reply qiskit/_instruction.py]: I'll make a separate issue for this to discuss. Here I didn't change the existing way.

File: test/python/test_circuit.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaygambetta) [reply test/python/test_circuit.py]: you had a test failing just fixed. If it passes I will merge

**Pull request merged**